packages/web-forms/src/lib/shared-state/form-state.ts

```diff
@@ -0,0 +1,117 @@
+import type {
+	CreatedFormInstance,
+	FormInstanceConfig,
+	LoadFormOptions,
+	LoadFormResult,
+	LoadFormSuccessResult,
+	LoadFormWarningResult,
+	RestoredFormInstance,
+} from '@getodk/xforms-engine';
+import { loadForm } from '@getodk/xforms-engine';
+import type { Ref } from 'vue';
+import { computed, reactive, ref } from 'vue';
+import type { AnyInstancePayload } from './instance-cache-state.ts';
+
+// prettier-ignore
+export type InstantiableFormResult =
+	| LoadFormSuccessResult
+	| LoadFormWarningResult;
+
+export const formResultState: Ref<LoadFormResult | null> = ref(null);
+
+const isInstantiableFormResult = (
+	formResult: LoadFormResult | null
+): formResult is InstantiableFormResult => {
+	return formResult != null && formResult.status !== 'failure';
```

Comment:
Fix:

This status ('failure') is safer coming from the xforms-engine, one source for these string constants. For example:

```suggestion
	return formResult != null && formResult.status !== FORM_STATE_STATUS_FAILURE;
```

I know we said about doing it at once in another PR, but if this is a low-hanging fruit, it's worth doing it now.
Since this status is being used outside the engine, I feel extra cautious

Reply:
I wouldn't be super concerned about addressing this case in this PR if that turns out to be important. However...

- We already merged the [engine side](https://github.com/getodk/web-forms/blob/faa81e48f9a2959a815ee8f1c7e3e22cbec07482/packages/xforms-engine/src/client/form/LoadFormResult.ts#L10-L14) in #335 

- As I mentioned [above](https://github.com/getodk/web-forms/pull/345#discussion_r1997401043), I'm not sure we want to land this demo in its current state anyway

- I'm _much more_ concerned that this suggestion of a constant doesn't reflect the approach we decided to take when we discussed it a couple days ago, nor #338 which captured that decision. I get the sense that we may need to revisit that decision. Which is probably all the more reason not to be hasty about adopting a new approach in this PR.

- I'm **even more** concerned about the impact that continued and growing focus on code style during review is having on productivity, and may have on _functionality_. The former is something I think we've all been feeling in one way or another. The latter is always a concern: in my experience, review focused on code style very often leads to more mistakes in behavior/approach slipping through, sometimes fairly big ones with significant impact. This is a big part of why I advocate for automated tools as a primary solution: I've seen the before/after, and it's pretty stark!

Reply:
Thanks for testing the assumptions. It gives reassurance

 >  think it's a lot more likely that we'll develop other clients before anyone else even wants to

Really looking forward to this!